src/components/typography/Heading.tsx

```diff
@@ -0,0 +1,52 @@
+import { Text, StyleSheet } from "react-native";
+
+import { global } from "@/constants/DesignTokens";
+
+type HeadingProps = {
+  size?:
+    | "heading1"
+    | "heading2"
+    | "heading3"
+    | "heading4"
+    | "heading5"
+    | "heading6";
+};
+
+type TextProps = HeadingProps & Text["props"];
+
+export default function Heading(props: TextProps) {
+  const { style, size = "heading1", ...otherProps } = props;
+
+  return <Text style={[styles.heading, styles[size], style]} {...otherProps} />;
+}
```

Comment:
Why are you doing it like that if you can destructure the same way inside the props?
```suggestion
export default function Heading({ style, size = "heading1", ...otherProps }: TextProps) {
  return <Text style={[styles.heading, styles[size], style]} {...otherProps} />;
}
```

Reply:
Changed it.